File: lian/week6/8-2579.py
import sys
input = sys.stdin.readline
n = int(input())
dp = [0] * 301
ls = [0] * 301
for i in range(1,n+1):
    ls[i] = int(input())

dp[1] = ls[1]
dp[2] = ls[1] + ls[2]
dp[3] = max(ls[1]+ls[3], ls[2]+ls[3])

for i in range(4, n + 1):
    dp[i] = max(dp[i - 2] + ls[i], dp[i - 3] + ls[i - 1] + ls[i])

print(dp[n])

File: leo/week6/8-2579.js
// Run by Node.js
const readline = require("readline");

(async () => {
  const rl = readline.createInterface({ input: process.stdin });
  let input = [];
  for await (const line of rl) {
    input.push(line.trim());
  }
  const n = parseInt(input[0]);
  const score = [0];
  for (let i = 1; i <= n; i++) {
    score.push(parseInt(input[i]));
  }
  const dp = Array.from({ length: n + 1 }, () => [0, 0]);
  dp[1][0] = score[1];
  dp[1][1] = score[1];
  for (let i = 2; i <= n; i++) {
    dp[i][0] = Math.max(dp[i - 2][0], dp[i - 2][1]) + score[i];
    dp[i][1] = dp[i - 1][0] + score[i];
  }
  const result = Math.max(dp[n][0], dp[n][1]);
  console.log(result);
})();
